src/main/java/frc/robot/subsystems/drivetrain/SwerveModule.java

```diff
@@ -0,0 +1,183 @@
+
+        angleMotor.selectProfileSlot(0, 0);
+        driveMotor.selectProfileSlot(1, 0);
+
+        this.driveMotor = driveMotor;
+        this.angleMotor = angleMotor;
+        this.wheel = wheel;
+    }
+
+
+    /**
+     * @return the speed of the wheel in [m/s]
+     */
+    public double getSpeed() {
+        return unitDrive.toVelocity(driveMotor.getSelectedSensorVelocity(1));
+    }
+
+    /**
+     * @return the angle of the wheel in radians
+     */
+    public double getAngle() {
+        return unitAngle.toUnits(angleMotor.getSelectedSensorPosition() - Constants.SwerveModule.ZERO_POSITION[wheel]);
+    }
+
+    /**
+     * sets the speed of the the wheel in ticks per 100ms
+     *
+     * @param speed the speed of the wheel in [m/s]
+     */
+    public void setSpeed(double speed) {
+        driveMotor.set(ControlMode.Velocity, unitDrive.toTicks100ms(speed));
+    }
+
+    /**
+     * sets the angle of the wheel, in consideration of the shortest path to the target angle
+     *
+     * @param angle the target angle in radians
+     */
+    public void setAngle(double angle) {
+        double targetAngle = getTargetAngle(angle, getAngle());
+        //int targetTicks = getTargetTicks(angle);
+        int angleTicks = unitAngle.toTicks(targetAngle) + Constants.SwerveModule.ZERO_POSITION[wheel];
+        angleMotor.set(ControlMode.Position, angleTicks);
+    }
+
+    /**
+     * finds the target angle of the wheel based on the shortest distance from the current position
+     *
+     * @param angle        the current target angle
+     * @param currentAngle the current angle of the wheel
+     * @return the target angle
+     */
+    public static double getTargetAngle(double angle, double currentAngle) {
+        // makes sure the value is between -pi and pi
+        angle = Utils.floorMod(angle, 2 * Math.PI);
+        double[] angles = {angle - 2 * Math.PI, angle, angle + 2 * Math.PI}; // An array of all possible target angles
+        double targetAngle = currentAngle;
+        double shortestDistance = Double.MAX_VALUE;
+        for (double target : angles) { // for each possible angle
+            if (Math.abs(target - currentAngle) < shortestDistance) // if the calculated distance is less than the current shortest distance
+            {
+                shortestDistance = Math.abs(target - currentAngle);
+                targetAngle = target;
+            }
+        }
+
+        return targetAngle;
+    }
+
+    /**
+     * stops the angle motor
+     */
+    public void stopAngleMotor() {
+        angleMotor.set(ControlMode.PercentOutput, 0);
+    }
+
+    /**
+     * reset encoder value to 0
+     */
+    public void resetAngle() {
+        angleMotor.setSelectedSensorPosition(0);
+    }
+
+    /**
+     * set the drive power
+     *
+     * @param power a value between [-1, 1]
+     */
+    public void setPower(double power) {
+        driveMotor.set(ControlMode.PercentOutput, power);
+    }
```

Comment:
you don't use this function, and if you are going to use it you need to rename the function, its unclear whether you apply power on the drive motor or the angle motor